mdata/store_cassandra.go

```diff
@@ -204,9 +204,16 @@ func (c *cassandraStore) processWriteQueue(queue chan *ChunkWriteRequest, meter
 
 			data := cwr.chunk.Series.Bytes()
 			chunkSizeAtSave.Value(int64(len(data)))
-			version := chunk.FormatStandardGoTsz
+			version := chunk.FormatStandardGoTszWithSpan
 			buf := new(bytes.Buffer)
-			binary.Write(buf, binary.LittleEndian, uint8(version))
+			binary.Write(buf, binary.LittleEndian, chunk.Format(version))
+
+			chunkSize, ok := chunk.RevChunkSpans[cwr.span]
+			if !ok {
+				// it's probably better to panic than to persist the chunk with a wrong length
+				panic(fmt.Sprintf("Chunk size invalid: %d", cwr.span))
```

Comment:
I can only think of 2 cases where this can happen:
* a critical bug in MT that would corrupt our data if MT kept running
* cosmic x-rays flipping bits in RAM or the cpu.

Neither should be attempted to be recovered in an automatic way IMHO (other than spinning up a new MT instance perhaps)